### ISSUE TITLE
--skip-solid doesn't apply with key  --devcontainer

### ISSUE THREAD

**Issue by alec-c4** [rails/rails#53482]
### Steps to reproduce

When I do create a new app with `--devcontainer` key, then another key `--skip-solid` doesn't work at all and app is created with solid-gems

```bash
# rails new app_name --skip-solid --devcontainer
```

### Expected behavior

No solid-gems in Gemfile, no solid schemas in `db` folder

### Actual behavior

Gems and schemas added.

### System configuration
**Rails version**:

8.0.0.rc1

**Ruby version**:

ruby 3.3.5 (2024-09-03 revision ef084cc8f4) [arm64-darwin24]

**Comment by vinibispo** [comment]
@alec-c4 I wasn't able to reproduce the issue
These were my steps
```bash
gem install rails -v 8.0.0.rc1
rails _8.0.0.rc1_ new app_name --skip-solid --devcontainer
```

**Comment by alec-c4** [comment]
@vinibispo could you please check config/database.yml for those lines:

```yaml
  cache:
    <<: *primary_production
    database: app_name_production_cache
    migrations_paths: db/cache_migrate
  queue:
    <<: *primary_production
    database: app_name_production_queue
    migrations_paths: db/queue_migrate
  cable:
    <<: *primary_production
    database: app_name_production_cable
    migrations_paths: db/cable_migrate
```

**Comment by vinibispo** [comment]
There's nothing on these lines, this is my database.yml
```yaml
# SQLite. Versions 3.8.0 and up are supported.
#   gem install sqlite3
#
#   Ensure the SQLite 3 gem is defined in your Gemfile
#   gem "sqlite3"
#
default: &default
  adapter: sqlite3
  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
  timeout: 5000

development:
  <<: *default
  database: storage/development.sqlite3

# Warning: The database defined as "test" will be erased and
# re-generated from your development database when you run "rake".
# Do not set this db to the same as development or production.
test:
  <<: *default
  database: storage/test.sqlite3


# Store production database in the storage/ directory, which by default
# is mounted as a persistent Docker volume in config/deploy.yml.
production:
  <<: *default
  database: storage/production.sqlite3
```

**Comment by alec-c4** [comment]
@vinibispo may I ask you to test with postgres?

**Comment by vinibispo** [comment]
Sure

**Comment by alec-c4** [comment]
Hmmm ... very strange - I've tried several times with something like 
```
rails new app_name -d postgresql -a propshaft -T --skip-system-test --skip-solid --devcontainer -j esbuild --master -c tailwind
```

and there was Gemfile with solid gems, database.yml with solid config and solid schemas. I'll try again with different configurations to reproduce this issue

**Comment by JoeDupuis** [comment]
Found the issue, the app generator is calling the devcontainer generator without passing the skip_solid option. 
The devcontainer generator override certain files when using postgres (like `database.yml`).

Fix is here: #53488
